float32 dx
float32 dy
float32 dtheta #+ve anticlockwise
uint32 scan_id
